lua/markview/integrations.lua

```diff
@@ -17,6 +17,43 @@ integrations.register_blink_source = function ()
 		ignore_enable = true
 	});
 
+	--- Dynamic filetype.
+	---@param filetype string
+	---@param config table | fun(): table See |blink-cmp-config-sources|.
+	local function handle_filetype(filetype, config)
+		---|fS
+
+		if type(config) == "table" and vim.islist(config) then
```

Comment:
It turns out that `per_filetype[filetype]` can be something like `{ inherit_defaults = false, "buffer" }`. The `inherit_defaults` key makes it so `vim.islist(config)` is `false`. This means the code will later assume this is a function and try to call it, which will obviously fail.

Reply:
Relevant config link:

https://cmp.saghen.dev/configuration/sources.html#providers